component/loki/source/kafka/kafka.go

```diff
@@ -53,7 +53,7 @@ type KafkaAuthentication struct {
 type KafkaSASLConfig struct {
 	Mechanism   string            `river:"mechanism,attr,optional"`
 	User        string            `river:"user,attr,optional"`
-	Password    string            `river:"password,attr,optional"`
+	Password    flagext.Secret    `river:"password,attr,optional"`
```

Comment:
Since we don't expose this setting as a flag, I think it should be set as a `rivertypes.Secret` type.

Sorry in advance for the confusion; initially the code was forklifted off Promtail, but we're gradually trying to make it more suited to Flow, and this is one of those cases.

Reply:
`SASLConfig` uses `flagext.Secret`. Should we be consistent with the secret types?

Reply:
IMHO, fields with River tags should use `rivertypes`; then we can convert to whatever format the underlying code requires.

In this case, it would work either way because flagext.Secret is just an embedded string, but in the general case it wouldn't be certain that Password could be decoded directly from the River type. WDYT?

Reply:
Yes, Previously I modified the `Password` field earlier to `flagext.Secret`, but as pointed earlier by @tpaschalis, I need to migrate it to rivertypes.Secret instead

Reply:
I was implying to change the type in the other struct as well, sorry. Maybe needs to be done in a separate PR